src/game_api/rpc.cpp

```diff
@@ -1846,3 +1847,25 @@ void game_log(std::string message)
     const static auto log_stream = (std::ofstream*)read_i64(get_address("game_log_stream"));
     game_log_fun(log_stream, message.c_str(), nullptr, LogLevel::Info);
 }
+
+using OnStateUpdate = void(StateMemory*);
+OnStateUpdate* g_state_update_trampoline{nullptr};
+void StateUpdate(StateMemory* s)
+{
+    g_state_update_trampoline(s);

```

Comment:
Could you add a pre-update callback too (PRE_EVERYFRAME?) that can cancel the trampoline by returning true (like many other callbacks), or I can do it after this is merged. I'm not sure why yet, but I think it might be useful.

Reply:
Done